File: GenericOOP.java
import java.lang.*;

class Arithematic<T>
{
    public T No1;
    public T No2;

    Arithematic(T a, T b)
    {
        No1 = a;
        No2 = b;
    }

    public T Addition()
    {
        if((No1 instanceof Integer) && (No2 instanceof Integer))
        {
            return (T)(Integer)((Integer)No1 + (Integer)No2);
        }
        else if((No1 instanceof Float) && (No2 instanceof Float))
        {
            return (T)(Float)((Float)No1 + (Float)No2);
        }
        else if((No1 instanceof Double) && (No2 instanceof Double))
        {
            return (T)(Double)((Double)No1 + (Double)No2);
        }
        else
        {
            return null;
        }
    }
}

class GenericOOP
{
    public static void main(String arg[])
    {
        Arithematic <Integer>obj = new Arithematic<Integer>(10,11);
        Integer ret = 0;
        ret = obj.Addition();
        System.out.println("Addition is : "+ret);
    }
}

File: GenericOOP.cpp
#include<iostream>
using namespace std;

template <class T>
class Arithematic
{
    public:
        T No1;
        T No2;

        Arithematic(T a, T b)
        {
            No1 = a;
            No2 = b;
        }

        T Addition()
        {
            T ans;
            ans = No1+No2;
            return ans;
        }
        T Substraction()
        {
            T ans;
            ans = No1-No2;
            return ans;
        }
};
int main()
{
    Arithematic <int>iobj(10,11);
    int ret = iobj.Addition();
    cout<<"Addition is : "<<ret<<"\n";

    Arithematic <double>dobj(10.9,11.5);
    double dret = dobj.Addition();
    cout<<"Addition is : "<<dret<<"\n";

    return 0;
}